src/main/java/sim/coffee/Server.java

```diff
@@ -1,14 +1,19 @@
 package sim.coffee;
 
-public class Server implements Runnable {
+public class Server implements Runnable{
+    private SharedQueue queue;
 
-    public Server(SharedQueue orders) {
+    public Server(SharedQueue queue) {
+        this.queue = queue;
     }
 
-    @Override
     public void run() {
-        // TODO Auto-generated method stub
-
+        while (!queue.getDone()) {
```

Comment:
So this is correct, but I think further to you comment above it should be `while "not done" and "queue isn't empty"`

Producer marks done once all customers have been put into the queue, then servers still need to serve any remaining after that

This may require a new method in Queue if there's not already one to check if it's empty.